support/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/FilePathPickle.java

```diff
@@ -86,6 +86,18 @@ protected FilePath tryResolve() {
         // TODO better to use a synchronized accessor
         @Restricted(NoExternalUse.class)
         public static final Map<VirtualChannel,String> channelNames = new WeakHashMap<VirtualChannel,String>();
+
+        static {
+            // Add channels that may already be known to Jenkins. This can be the case
```

Comment:
> how it's more robust

Handles the case that `onOnline` was not delivered for some reason.

> iterates the list of `Computer`s directly

See JENKINS-26096 for why that does not work.